### DISCUSSION TITLE
Adding an array select to a standard Map Index

### DISCUSSION THREAD

**Discussion by asapostolov** [ravendb/ravendb#13738] [Q&A] [answered]
I have the following query.
``` C#
RavenSession.Query<IncomeRow>()
                  .Customize( x => x.WaitForNonStaleResults() )
                  .Where( x => x.Payments.Any( z => z.DocumentId.In( documentIds ) ) )
                  .ToList()
```

This creates an Auto index with mapped property that looks like this `Payments[].DocumentId` and the query works as expected.

How can I add such property to a standard index Map definition? 
I tried with `row.Payments.Select(x=>x.DocumentId)` in the Map definition but I'm not sure how to write the left part of that - the name of the property, so it doesn't compile.

**Comment by arekpalinski** [comment, marked as answer]
If you put a collection into an indexed field then you'll be able to find it if any item matches there. Sample index def:

```
from row in docs.IncomeRows select new {
    
    Payments_DocumentId = row.Payments.Select(x => x.DocumentId)
}
```

RQL query:

```
from index 'test' where Payments_DocumentId in ("docs/1") 
```

Similar sample you an find here: https://ravendb.net/docs/article-page/5.3/csharp/indexes/querying/filtering#where--containsany

**Reply by asapostolov** [reply]
Yes, but I want to use the original model of the document in C# in order to query the index with LINQ e.g. `.Where( x => x.Payments.Any( z => z.DocumentId.In( documentIds ) ) )`.

So is there any way to define a property that looks like this `Payments[].DocumentId` in a C# Map Index definition?

**Reply by arekpalinski** [reply]
Also please have a look at the index definition in the following example when using Client API:
https://ravendb.net/docs/article-page/5.3/csharp/client-api/session/querying/how-to-project-query-results#oftype-(as)---simple-projection

Note that the shape of index entry might be different than a document so you use a different type when querying and then getting back to entity type using `OfType()`

https://ravendb.net/docs/article-page/5.3/csharp/client-api/session/querying/how-to-project-query-results#oftype-(as)---simple-projection

**Reply by arekpalinski** [reply]
I modified the answer and used `Payments_DocumentId` instead of `DocumentId`. See here for explanation how `_` will be added automatically to the property name when querying:

https://ravendb.net/docs/article-page/5.3/csharp/client-api/configuration/querying#findpropertynameforindex-and-findpropertynamefordynamicindex

**Reply by asapostolov** [reply]
`Payments_DocumentId` works! Thanks a lot!
